### PR TITLE
servant-client-core: Redact Authorization header

### PR DESCRIPTION
**Pull request by roberth**
Prevent credentials leaking in exception messages.

### CODE REVIEW

File: servant-client-core/src/Servant/Client/Core/Request.hs [hunk 1]
```diff
@@ -64,8 +64,32 @@ data RequestF body path = Request
...
   , requestMethod      :: Method
-  } deriving (Generic, Typeable, Eq, Show, Functor, Foldable, Traversable)
+  } deriving (Generic, Typeable, Eq, Functor, Foldable, Traversable)
 
+instance (Show a, Show b) =>
+           Show (Servant.Client.Core.Request.RequestF a b) where
+    showsPrec p req
+      = showParen
+        (p >= 11)
+        ( showString "Request {requestPath = "
```
Comment (phadej) [inline servant-client-core/src/Servant/Client/Core/Request.hs]: `$`

**Comment by roberth** [comment]
I completely missed the final comment on the review.

**Comment by domenkozar** [comment]
We published the advisory for cachix: https://github.com/cachix/cachix/security/advisories/GHSA-5v3x-gf4h-9qrh

**Comment by nh2** [comment]
For my record, these are the relevant `nixpkgs` commits (thanks @domenkozar):

* `master`: https://github.com/NixOS/nixpkgs/commit/f1682a7f126d4d56dfbb96bb8c8c5582abb22828
* `19.09`: https://github.com/NixOS/nixpkgs/commit/195cbccf3f408b4209144d0d8e2847e7fa532402

**Pull request merged**